Mathlib/Algebra/Algebra/Operations.lean

```diff
@@ -473,6 +473,7 @@ protected theorem pow_induction_on_left' {C : ∀ (n : ℕ) (x), x ∈ M ^ n →
       (fun x hx y hy Cx Cy => hadd _ _ _ _ _ Cx Cy) hx
 #align submodule.pow_induction_on_left' Submodule.pow_induction_on_left'
 
+set_option maxHeartbeats 700000 in
```

Comment:
I expect I'll be able to drop this by refactoring the proof below but first I want to see how CI reacts to the rest of the proposed changes.